### DISCUSSION TITLE
[Feature] alternative reverse geocoding implementation

### DISCUSSION THREAD

**Discussion by jkasten80** [immich-app/immich#2684] [Feature Request]
### The feature

The geocoding feature over geonames.org is not very accurate. It would be great to have the possebility to use google maps api for geocoding. It would be nice to put my own google api key in and use it for geocoding instead of geonames.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#11486

**Comment by MichelZ** [comment]
I've built exactly this now, it uses overture maps and uses a place's geometry/bounding box, not just a point on the map to calculate the place: 
Docs: https://immich-reversegeo.github.io/
GitHub: https://github.com/immich-reversegeo/immich-reversegeo
